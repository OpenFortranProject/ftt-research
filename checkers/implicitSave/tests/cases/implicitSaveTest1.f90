module test
  implicit none
  contains
  subroutine f
    integer :: x = 4
    print *, x
    x = x + 1
  end subroutine f
end module

program main
  use test
  print *, "Testing f"
  call f
  call f
  call f
end program main
