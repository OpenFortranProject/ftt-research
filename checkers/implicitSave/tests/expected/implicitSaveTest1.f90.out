Warning: environment variable COMPASS_PARAMETERS is not set, trying to use compass_parameters of the current path
 Running Prerequisite SgProject
ImplicitSave: /home/dagit/research/checkers/implicitSave/tests/cases/implicitSaveTest1.f90:4.2: improper operation to an implicit saved variable:expression 'x = x + 1' value re-assigned
